core/play/src/main/scala/play/api/controllers/Assets.scala

```diff
@@ -821,7 +832,7 @@ class AssetsBuilder(errorHandler: HttpErrorHandler, meta: AssetsMetadata) extend
       case Some((assetInfo, acceptEncoding)) =>
         val connection = assetInfo.url(acceptEncoding).openConnection()
         // Make sure it's not a directory
-        if (Resources.isUrlConnectionADirectory(connection)) {
+        if (env != null && Resources.isUrlConnectionADirectory(env.classLoader, connection)) {
```

Comment:
For backwards compatibility, shouldn't we keep calling `Resources.isUrlConnectionADirectory` even when env is null?
```suggestion
        if (Resources.isUrlConnectionADirectory(if (env != null) env.classLoader else null, connection)) {
```
Makes sense IMHO.

Reply:
I see. Certainly, even if it already exists, there is a path in the calling hierarchy where classLoader is assumed to be null. I will reflect your suggestions.